embassy-stm32/src/exti.rs

```diff
@@ -250,14 +215,30 @@ impl_exti!(EXTI13, 13);
 impl_exti!(EXTI14, 14);
 impl_exti!(EXTI15, 15);
 
-/// safety: must be called only once
-pub(crate) unsafe fn init() {
-    interrupt::EXTI0::steal().enable();
-    interrupt::EXTI1::steal().enable();
-    interrupt::EXTI2::steal().enable();
-    interrupt::EXTI3::steal().enable();
-    interrupt::EXTI4::steal().enable();
-    interrupt::EXTI9_5::steal().enable();
-    interrupt::EXTI15_10::steal().enable();
-    interrupt::EXTI15_10::steal().enable();
+macro_rules! impl_exti_irq {
+    ($e:ident) => {
+        #[interrupt]
+        unsafe fn $e() {
+            crate::exti::on_irq()
+        }
+    };
+
+    ($e:ident, $($es:ident),+) => {
+        impl_exti_irq! { $e }
+        impl_exti_irq! { $($es),+ }
+    };
+}
+
+macro_rules! impl_exti_init {
+    ($e:ident) => {
+        crate::interrupt::$e::steal().enable();
+    };
+
+    ($e:ident, $($es:ident),+) => {
+        /// safety: must be called only once
+        pub(crate) unsafe fn init() {
```

Comment:
This macro doesn't work, it only enables the first irq because the rest is emitted in another nested `init` fn: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a64213eb9a462f7ed9f00ff1a310d4f2

using macro repetition directly is easier:

```rust


macro_rules! impl_exti_init {
    ($($e:ident),+) => {
        /// safety: must be called only once
        pub(crate) unsafe fn init() {
            $(
                crate::interrupt::$e::steal().enable();
            )+
        }
    };
}

Reply:
Hmm, I thought I had that fixed already :/ Agreed it makes sense to combine the macros and do it this way, macro skills++ :)